02-peer-protocol.md

```diff
@@ -1862,6 +1962,49 @@ it's simplest to only allow it to set fee levels; however, as the same
 fee rate applies to HTLC transactions, the receiving node must also
 care about the reasonableness of the fee.
 
+### Updating Block Height: `update_blockheight` (`option_will_fund`)
+
+An `update_blockheight` message is sent by initiator node.
+Like any update, it's first committed to the receiver's commitment
+transaction and then (once acknowledged) committed to the sender's. Unlike
+an HTLC, `update_blockheight` is never closed but simply replaced.
+
+No `update_blockheight` changes may be pending when sent (i.e. must be
+fully committed on both sides).
+
+1. type: 137 (`update_blockheight`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u32`:`blockheight`]
+
+#### Requirements
+
+The node _responsible_ for initiating the channel:
+  - MUST send `update_blockheight` to ensure the CSV lock mandated by
+    `option_will_fund` will expire at the end of the funding lease.
+  - MUST send `update_blockheight` when their local blockheight updates
+
+The node _not responsible_ for initiating the channel:
+  - MUST NOT send `update_blockheight`.
+
+A receiving node:
+  - if the `update_blockheight` is less than the last received `blockheight`:
+    - SHOULD fail the channel.
+  - if the sender is not the initiator:
+    - MUST fail the channel.
+  - if `blockheight` is more than 504 blocks behind the current blockheight:
+    - SHOULD fail the channel
```

Comment:
Not quite right, this needs a separate section (hmm, may need an intro which defines lease nomenclature?  Lessor, end of lease, etc?)

A node which offered a lease on a channel, which is still being enforced:
- if the last received `blockheight` is more than 504 blocks behind the current blockheight:
  - SHOULD fail the channel.

With rationale:

The lessor's lease is only shortened when a new blockheight is committed, otherwise it's always 4032 blocks in the future. If the peer doesn't update the blockheight, at some point it has to start the closing process to retrieve its funds.  504 blocks seems a reasonable (but not compulsory!) answer here.